### PR TITLE
Add ROSA support for Tempo operator installation 

### PR DESCRIPTION
**Pull request by Sandeep20013**
# Pull Request

## Summary

- Added patch_tempo_operator_for_rosa function to configure Tempo operator with AWS IAM Role ARN for ROSA clusters. The installation now validates that TEMPO_ROLE_ARN environment variable is set before proceeding, providing clear error messaging if missing. 

- Fixed test_guardrails.py by adding missing minio_pod and minio_data_connection fixture parameters.

## Related Issues

- JIRA: https://redhat.atlassian.net/browse/RHOAIENG-73367 https://redhat.atlassian.net/browse/RHOAIENG-60312

## Please review and indicate how it has been tested

- [x] Locally
- [ ] Jenkins

## Summary by CodeRabbit

* **Tests**
  * Expanded AI safety guardrails coverage for MinIO-backed Hugging Face detector configurations.
  * Enhanced Tempo operator validation to confirm the expected role ARN is applied to every container.
  * Improved test reliability with required role configuration, deployment reconciliation checks, and clear timeout failures.

### CODE REVIEW

File: tests/ai_safety/guardrails/utils.py [hunk 1]
```diff
@@ -14,6 +15,28 @@
...
+    containers = deployment_dict["spec"]["template"]["spec"]["containers"]
+
+    for container in containers:
+        if "env" not in container:
+            container["env"] = []
+        # Add ROLEARN environment variable
+        container["env"].append({"name": "ROLEARN", "value": role_arn})
+
+    deployment.update(deployment_dict)
+    deployment.wait_for_replicas()
```
Comment (dbasunag) [inline tests/ai_safety/guardrails/utils.py:73]: You may hit a corner case here. Where you wait for replicas call prematurely says it is done with patching. I would suggest, checking your deployments status conditions first and ensure that the last transition time before the update was changed first. e.g. 
```
- lastTransitionTime: "2026-07-09T21:03:19Z"
    lastUpdateTime: "2026-07-09T21:03:19Z"
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
```.
@sheltoncyril did something similar a few months back. So please check with him how to do this.

File: tests/ai_safety/guardrails/conftest.py [hunk 2]
```diff
@@ -267,6 +269,14 @@ def installed_tempo_operator(admin_client: DynamicClient, model_namespace: Names
     tempo_operator_subscription = Subscription(client=admin_client, namespace=operator_ns.name, name=package_name)
 
     if not tempo_operator_subscription.exists:
+        # Check if TEMPO_ROLE_ARN is required but not set
```
Comment (sheltoncyril) [inline tests/ai_safety/guardrails/conftest.py:272]: An AI-assisted review flagged this and I wanted to check before treating it as actionable.

The ROSA handling (require `TEMPO_ROLE_ARN`, patch `ROLEARN`, log "ROSA cluster detected") seems to run unconditionally with no ROSA/platform detection, which looks like it would break plain-OCP Tempo installs. Also the `starting_csv` pin was removed, which could let the Tempo version float. Are these intended for all clusters, or should the ROSA path be gated? Thanks!
